src/main/java/com/iexec/core/replicate/ReplicatesService.java

```diff
@@ -224,6 +220,97 @@ private boolean isStatusBeforeWorkerLostEqualsTo(Replicate replicate, ReplicateS
+            log.error("Cannot update replicate, could not get replicate [chainTaskId:{}, UpdateRequest:{}]",
+                    chainTaskId, statusUpdate);
+            return ReplicateStatusUpdateError.UNKNOWN_REPLICATE;
+        }
+
+        ReplicatesList replicatesList = oReplicateList.get();
+        Replicate replicate = replicatesList.getReplicateOfWorker(walletAddress).orElseThrow(); // "get" could be used there but triggers a warning
+        ReplicateStatus newStatus = statusUpdate.getStatus();
+
+        boolean hasAlreadyTransitionedToStatus = replicate.containsStatus(newStatus);
+        if (hasAlreadyTransitionedToStatus) {
+            log.warn("Cannot update replicate, status {} already reported.", newStatus);
+            return ReplicateStatusUpdateError.ALREADY_REPORTED;
+        }
+
+        boolean isValidTransition = ReplicateWorkflow.getInstance()
+                .isValidTransition(replicate.getCurrentStatus(), newStatus);
+        if (!isValidTransition) {
+            log.warn("Cannot update replicate, bad workflow transition {}",
+                    getStatusUpdateLogs(chainTaskId, replicate, statusUpdate));
+            return ReplicateStatusUpdateError.BAD_WORKFLOW_TRANSITION;
+        }
+
+        boolean canUpdate = true;
+
+        switch (newStatus) {
+            case CONTRIBUTE_FAILED:
+            case REVEAL_FAILED:
+                canUpdate = false;
+                break;
+            case RESULT_UPLOAD_FAILED:
+                canUpdate = verifyStatus(chainTaskId, walletAddress, newStatus);
+                break;
+            case CONTRIBUTED:
+            case REVEALED:
+                canUpdate = canUpdateToBlockchainSuccess(chainTaskId, replicate, statusUpdate, updateReplicateStatusArgs);
+                break;
+            case RESULT_UPLOADED:
+                canUpdate = canUpdateToUploadSuccess(chainTaskId, replicate, statusUpdate, updateReplicateStatusArgs);
+                break;
+            default:
+                break;
+        }
+
+        if (!canUpdate) {
+            log.warn("Cannot update replicate {}",
+                    getStatusUpdateLogs(chainTaskId, replicate, statusUpdate));
+            return ReplicateStatusUpdateError.GENERIC_CANT_UPDATE;
+        }
+
+        return ReplicateStatusUpdateError.NO_ERROR;
+    }
+
+    /**
+     * Computes arguments used by {@link ReplicatesService#updateReplicateStatus(String, String, ReplicateStatusUpdate, UpdateReplicateStatusArgs)}.
+     * These arguments can be null or empty, no check is done there.
+     */
+    public UpdateReplicateStatusArgs computeUpdateReplicateStatusArgs(String chainTaskId,
+                                                                      String walletAddress,
+                                                                      ReplicateStatusUpdate statusUpdate) {
+        int workerWeight = iexecHubService.getWorkerWeight(walletAddress);
```

Comment:
Contribution and worker weight are only required for some statuses

Reply:
That's fixed. Both of them are used only for `CONTRIBUTED` status.